src/metabase/models/collection.clj

```diff
@@ -1143,6 +1143,28 @@
         (assoc user :personal_collection_id (or (user-id->collection-id (u/the-id user))
                                                 (user->personal-collection-id (u/the-id user))))))))
 
+#_{:clj-kondo/ignore [:unused-private-var]}
+(mi/define-simple-hydration-method ^:private collection-is-personal
+  :is_personal
+  "Whether the current collection is or nested in a personal collection."
+  [collection]
+  (is-personal-collection-or-descendant-of-one? collection))
+
+(mi/define-batched-hydration-method collections-is-personal
+  :collections_is_personal
+  "Efficiently hydrate the `:is_personal` property of a sequence of Collections.
+  `true` means the collection is or nested in a personal collection."
+  [collections]
+  (let [personal-collection-ids (t2/select-pks-set :model/collection :personal_owner_id [:not= nil])
+        location-is-personal    (fn [location]
+                                  (boolean
+                                   (and (string? location)
+                                        (some #(str/starts-with? location (format "/%d/" %)) personal-collection-ids))))]
+    (map (fn [{:keys [location personal_owner_id] :as coll}]
+           (assoc coll :is_personal (or (some? personal_owner_id)
+                                        (location-is-personal location))))
+         collections)))
```

Comment:
You can define these with one `mi/define-batched-hydration-method` call, to avoid having two hydration keys for the same "thing".

```clojure
(mi/define-batched-hydration-method collections-is-personal
  :collections_is_personal
  "Efficiently hydrate the `:is_personal` property of a sequence of Collections.
  `true` means the collection is or nested in a personal collection."
  [collections]
  (if (= (count collections) 1)
    (is-personal-collection-or-descendant-of-one? collection)
    (let [personal-collection-ids (t2/select-pks-set :model/collection :personal_owner_id [:not= nil])
          location-is-personal    (fn [location]
                                    (boolean
                                     (and (string? location)
                                          (some #(str/starts-with? location (format "/%d/" %)) personal-collection-ids))))]
      (map (fn [{:keys [location personal_owner_id] :as coll}]
             (assoc coll :is_personal (or (some? personal_owner_id)
                                          (location-is-personal location))))
           collections))))
```

Reply:
Cool, I removed the simple hydration here [ed291d8](https://github.com/metabase/metabase/pull/34827/commits/ed291d859d82d81b071cd2f9770fcbcc9baef458)